torch_sim/models/fairchem.py

```diff
@@ -38,355 +38,359 @@
-                    fixed=fixed[c - n : c].clone(),
-                    natoms=n,
-                    pbc=torch.tensor([state.pbc, state.pbc, state.pbc], dtype=torch.bool),
+            natoms = torch.bincount(state.batch)
+            fixed = torch.zeros((state.batch.size(0), natoms.sum()), dtype=torch.int)
+            data_list = []
+            for i, (n, c) in enumerate(
+                zip(natoms, torch.cumsum(natoms, dim=0), strict=False)
+            ):
+                data_list.append(
+                    Data(
+                        pos=state.positions[c - n : c].clone(),
+                        cell=state.row_vector_cell[i, None].clone(),
+                        atomic_numbers=state.atomic_numbers[c - n : c].clone(),
+                        fixed=fixed[c - n : c].clone(),
+                        natoms=n,
+                        pbc=torch.tensor(
+                            [state.pbc, state.pbc, state.pbc], dtype=torch.bool
+                        ),
+                    )
                 )
+            self.data_object = Batch.from_data_list(data_list)
+
+            if self.dtype is not None:
+                self.data_object.pos = self.data_object.pos.to(self.dtype)
+                self.data_object.cell = self.data_object.cell.to(self.dtype)
+
+            predictions = self.trainer.predict(
+                self.data_object, per_image=False, disable_tqdm=True
             )
-        self.data_object = Batch.from_data_list(data_list)
 
-        if self.dtype is not None:
-            self.data_object.pos = self.data_object.pos.to(self.dtype)
-            self.data_object.cell = self.data_object.cell.to(self.dtype)
+            results = {}
 
-        predictions = self.trainer.predict(
-            self.data_object, per_image=False, disable_tqdm=True
-        )
+            for key in predictions:
+                _pred = predictions[key]
+                if key in self._reshaped_props:
+                    _pred = _pred.reshape(self._reshaped_props.get(key)).squeeze()
+                results[key] = _pred.detach()
 
-        results = {}
+            results["energy"] = results["energy"].squeeze(dim=1)
+            if results.get("stress") is not None and len(results["stress"].shape) == 2:
+                results["stress"] = results["stress"].unsqueeze(dim=0)
+            return results
 
-        for key in predictions:
-            _pred = predictions[key]
-            if key in self._reshaped_props:
-                _pred = _pred.reshape(self._reshaped_props.get(key)).squeeze()
-            results[key] = _pred.detach()
+except ImportError as exc:
 
-        results["energy"] = results["energy"].squeeze(dim=1)
-        if results.get("stress") is not None and len(results["stress"].shape) == 2:
-            results["stress"] = results["stress"].unsqueeze(dim=0)
-        return results
+    class FairChemModel(torch.nn.Module, ModelInterface):
```

Comment:
What if instead we throw a warning if this class is defined? That could allow for an easier time catching the bug without indenting the whole file.